tests/test_transformations.py

```diff
@@ -0,0 +1,127 @@
+def test_coordinate_window_shift(coordinate_pair, window_size, video_shape,
+                                 expected_pair):
+    transformed_center = transformations.get_transformed_center(coordinate_pair,
+                                                                     window_size,
+                                                                     video_shape)
+    assert transformed_center == expected_pair
+
+
+def test_mp4_conversion(video_fixture):
+    output_path = Path(__file__).parent
+    output_path = output_path / 'video.mp4'
+
+    fps = 30
+
+    transformations.transform_to_mp4(video=video_fixture,
+                                     output_path=output_path.as_posix(),
+                                     fps=fps),
+
+    frames = []
+    reader = imageio.get_reader(output_path.as_posix(), mode='I', fps=fps,
+                                size=(2, 2), pixelformat="gray")
+    for frame in reader:
+        gray_frame = cv2.cvtColor(frame, cv2.COLOR_RGB2GRAY)
+        frames.append(gray_frame)
+    frames = np.stack(frames)
+
+    # codec for mp4 encoding changes the video slightly between h5 and mp4
+    # and vice versa, so numbers are not exactly the same. Instead general
+    # structure of video array loaded. Cv color conversion does not rectify
+    # this error unfortunately, this has been attempted and small errors
+    # persist
+
+    assert frames.shape == video_fixture.shape
+
+    os.remove(output_path)
+
+
+@pytest.mark.parametrize("video,coord,box_size,expected",
+    [(test_video, (0, 0), (3, 3),
+      np.stack([np.array([[0, 1, 2],
+                          [5, 6, 7],
+                          [10, 11, 12]])]*2)),
+     (test_video, (2, 2), (3, 3), np.stack([np.array([[6, 7, 8],
+                                                      [11, 12, 13],
+                                                      [16, 17, 18]])]*2)),
+     (test_video, (-1, -1), (3, 3), np.stack([np.array([[0, 1, 2], [5, 6, 7],
+                                                       [10, 11, 12]])]*2)),
+     (test_video, (10, 10), (3, 3), np.stack([np.array([[12, 13, 14], [17, 18, 19],
+                                                        [22, 23, 24]])] * 2)),
+     (test_video, (0, 10), (3, 3), np.stack([np.array([[10, 11, 12], [15, 16, 17],
+                                                       [20, 21, 22]])] * 2)),
+     (test_video, (10, 0), (3, 3), np.stack([np.array([[2, 3, 4], [7, 8, 9],
+                                                       [12, 13, 14]])] * 2)),
+     (test_video, (2, -1), (3, 3), np.stack([np.array([[1, 2, 3], [6, 7, 8],
+                                                        [11, 12, 13]])] * 2)),
+     (test_video, (-1, 2), (3, 3), np.stack([np.array([[5, 6, 7], [10, 11, 12],
+                                                       [15, 16, 17]])]*2)),
+     (test_video, (0, 0), (5, 5), test_video),
+     (test_video, (2, 2), (0, 0), np.zeros(shape=(2, 0, 0)))]
```

Comment:
What if the box is too big for the video?

Reply:
Raises an exception in the function for centered coordiantes, I'm not sure how to test for a raised exception.

Reply:
Here's an example:
https://github.com/AllenInstitute/segmentation-labeling-app/blob/8abfdb22845bca4fd01c28255d6c115d16172a8b/tests/test_transfer_utils.py#L83-L84
and the docs:
https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Reply:
This doesn't explain how to use the pytest raise for only one parameterized case

Reply:
Also see https://github.com/AllenInstitute/segmentation-labeling-app/blob/master/tests/transforms/test_array_utils.py

Line 52 I have the separated test with a clear name to say its purpose.